integration/log_integration_test.go

```diff
@@ -452,3 +496,43 @@ func min(a, b int64) int64 {
 
 	return b
 }
+
+// compareLogAndTreeProof compares a proof received from the log against one generated by
+// an in memory merkle tree. It ensures the proofs contain an identical list of node hashes.
+func compareLogAndTreeProof(logProof *trillian.Proof, treeProof []merkle.TreeEntryDescriptor) error {
+	// Compare the proof lengths
```

Comment:
Idle thought: would it be simpler replace this fn with a comparison of the results of `formatLogProof` and `formatTreeProof` ?